kong/plugins/bot-detection/handler.lua

```diff
@@ -0,0 +1,71 @@
+local BasePlugin = require "kong.plugins.base_plugin"
+local reg = require "rex_pcre"
+local responses = require "kong.tools.responses"
+local rules = require "kong.plugins.bot-detection.rules"
+local stringy = require "stringy"
+local bot_cache = require "kong.plugins.bot-detection.cache"
+
+local ipairs = ipairs
+
+local BotDetectionHandler = BasePlugin:extend()
+
+BotDetectionHandler.PRIORITY = 2500
+
+local ALLOW = 1
+local BLOCK = 2
+
+function BotDetectionHandler:new()
+  BotDetectionHandler.super.new(self, "bot-detection")
+end
+
+function BotDetectionHandler:access(conf)
+  BotDetectionHandler.super.access(self)
+
+  local user_agent = ngx.req.get_headers()["user-agent"]
```

Comment:
> It could be that user_agent is a table here if the header is present multiple times.

This is a very common situation in many plugins - I wonder if we should start considering exporting some common functionality somewhere else. Many checks are redundant in many plugins. Also in a 1.0 scenario, we want people to leverage a common set of utils for the most common operations.